src/streamlink_cli/main.py

```diff
@@ -931,16 +931,11 @@ def main():
             # Close output
             if output:
                 output.close()
-
-            # Make sure current stream gets properly cleaned up
-            if stream_fd:
-                console.msg("Interrupted! Closing currently open stream...")
-                try:
-                    stream_fd.close()
-                except KeyboardInterrupt:
-                    sys.exit()
-            else:
-                console.msg("Interrupted! Exiting...")
+            console.msg("Interrupted! Exiting...")
+        finally:
+            if hasattr(stream_fd, "close"):
```

Comment:
If stream_fd is None hasattr returns False and close() will not be called. Is not it?